### PR TITLE
Fix axis label auto-rotation

### PR DESCRIPTION
**Pull request by tomassatinsky**
By copying and fixing Highcharts plugin grouped-categories

About the copy of grouped-categories plugin 

Kate Downing confirmed that
 - this is entirely under Creative Commons Attribution 3.0. This is fine to use, and modify, in any GoodData product. As with all third party code, though, it's important that it be kept distinct and apart from GoodData-written files so that when we run OSS scanning tools to create open source disclosures, we pick it up on our scans. If you have any questions on that, I think Vojtech Luhan can give extra direction.
 
Vojtech Luhan confirmed that
- it's OK to have the code just in separate file
- It's enough to include URI of the CC licence

JIRA: FET-617

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by tomassatinsky** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2620/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4253/ : FAILURE in 9m 35s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2621/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/700/ : FAILURE in 9m 35s

File: libs/sdk-ui-charts/src/highcharts/adapter/plugins/group-categories-wrapper.ts [hunk 1]
```diff
@@ -20,42 +18,4 @@ export function groupCategoriesWrapper(Highcharts: any): void {
 
         hideGridInXAxis.call(this);
     });
-
-    wrap(Highcharts.Tick.prototype, "getPosition", function (
```
Comment (tomassatinsky) [inline libs/sdk-ui-charts/src/highcharts/adapter/plugins/group-categories-wrapper.ts:24]: This was currently causing a bug. Previously it was probably fixing some issue that has been resolved in Highcharts already.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2638/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4270/ : FAILURE in 5m 39s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2640/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4272/ : FAILURE in 5m 17s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2641/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4273/ : SUCCESS in 22m 38s

**Comment by tomassatinsky** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/704/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2646/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/704/ : UNSTABLE in 20m 29s

File: libs/sdk-ui-charts/package.json [hunk 2]
```diff
@@ -65,7 +65,6 @@
         "custom-event": "^1.0.1",
         "date-fns": "^2.8.1",
         "highcharts": "8.2.2",
-        "highcharts-grouped-categories": "^1.1.6",
```
Comment (no23reason) [inline libs/sdk-ui-charts/package.json:68]: Please run `rush update` we should see some changes in `browser-approved-pacakges.json` as this was the only package using the plugin

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2649/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4280/ : SUCCESS in 15m 15s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2651/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4282/ : SUCCESS in 24m 14s

**Review by yanlipnican — APPROVED** [review]

**Comment by yanlipnican** [comment]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1081

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/793/ : FAILURE in 1m 51s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1057/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4285/ : SUCCESS in 16m 12s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/158/ : SUCCESS in 8m 07s (non-voting)

**Pull request merged**